### PR TITLE
Add Platforms Tags and Page

### PR DESCRIPTION
**Pull request by valencik**
**TLDR:**
- replaces `category` tag with `platform` tag
- page for each platform we support (JS, JVM, Native)

## Preview

While open, this PR is available for preview here:
https://valencik.github.io/typelevel.github.com/platforms/
(note however that some links will be broken there)

I recommend checkout out that preview because these screenshots are probably already out of data (they are).

Screenshot 2023-04-22 at 10-13-19 Typelevel Typelevel js

Screenshot 2023-04-22 at 10-13-30 Typelevel Typelevel Native

## Platform Pages

I would like for these pages to:
- talk a bit about the support and capabilities on each platform
  - e.g. the native page should link to [Arman's blog post](https://typelevel.org/blog/2022/09/19/typelevel-native.html)
  - e.g. the js page should briefly discuss the JS environment options: browser, severless, cli
-  list the projects that are supported by each platform

I want to be slightly more than just another project breakout list.
However, that might be achieved over multiple PRs so we can divide up the yak shaving.

## Tags

This PR also removes the `category` tag in favour of the `platforms` tag.
I am hoping this isn't too controversial.
I do not think the `category` tag is useful in its current form, a possibly more useful tag might pull in the GitHub topics, but that is out of scope here. The category tag is also not universally used, and it almost requires a bit of moderation. Whereas the `platform` tag is a lot more factual.

### CODE REVIEW

File: _data/description.yml [hunk 1]
```diff
@@ -4,6 +4,10 @@ aboutDescription: Learn more about the organization, including our main goals, c
 projectsTitle: Projects
 projectsDescription: Our projects cover a wide range of domains, from general functional programming to tooling.
 
+platformsTitle: Platforms
+platformsDescription: Checkout the Typelevel ecosystem by platform.
+jsDescription: Checkout Typelevel on ScalaJS!
```
Comment (armanbilge) [inline _data/description.yml:9]:
```suggestion
jsDescription: Checkout Typelevel on Scala.js!
```

Ref: https://github.com/scala-js/scala-js-macrotask-executor/pull/28#discussion_r730727662

File: _data/description.yml [hunk 2]
```diff
@@ -4,6 +4,10 @@ aboutDescription: Learn more about the organization, including our main goals, c
 projectsTitle: Projects
 projectsDescription: Our projects cover a wide range of domains, from general functional programming to tooling.
 
+platformsTitle: Platforms
+platformsDescription: Checkout the Typelevel ecosystem by platform.
+jsDescription: Checkout Typelevel on Scala.js!
```
Comment (jducoeur) [inline _data/description.yml:9]: I'm a little confused why this goes in a property -- is it used in more than one place?

But in general, all looks good to me!

Comment (valencik) [reply _data/description.yml:9]: I was kind of just copying what I saw. But I do believe I'm going to move it out into the html because I want the text to have some hyperlinks. And I feel like the html is the more "obvious" place to find it.
AND... as you point out, it likely will not be used in more than one place :)

**Comment by valencik** [comment]
Hey folks, I think this is finally ready for review.
You can preview the changes at: https://valencik.github.io/typelevel.github.com/platforms/js/

File: README.md [hunk 3]
```diff
@@ -132,7 +132,6 @@ To add a regular project, create a new markdown file in the `_projects` folder w
 ```yml
 layout: post
 title: "Cats"
-category: "Functional Programming"
```
Comment (samspills) [inline README.md:135]: Add the platforms line as part of the example?

I can't add it as a suggestion :( 
`platforms: [js, jvm, native]`

File: platforms/js.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+<div id="section-page">
+  <div class="container">
+    <div class="masthead-page">
+      <h1><span>{{ page.title }}</span></h1>
+      <p>
+        Much of the Typelevel ecosystem cross compiles to JavaScript via <a href="https://www.scala-js.org/">Scala.js</a>.
+        Deploy web servers on NodeJS with <a href="https://github.com/http4s/http4s">http4s</a>, <a href="https://github.com/typelevel/fs2">fs2</a>, and <a href="https://github.com/typelevel/cats-effect">cats-effect</a>.
+        Build reactive frontends with <a href="https://github.com/outwatch/outwatch">outwatch</a>, <a href="https://github.com/armanbilge/calico">calico</a>, and <a href="https://github.com/armanbilge/fs2-dom">fs2-dom</a>.
```
Comment (samspills) [inline platforms/js.html:14]: I really like these descriptions 🎉

File: README.md (hunk 3 above)
Comment (samspills) [reply README.md:135]: Oh closer inspection, this whole section might need a bit of a revamp. Line 128 is still referring to incubator projects, and the example also includes a `layout` key that I don't think is necessary.

**Review by samspills — APPROVED** [review]
I think this is great 🥇

I did leave a comment about the readme example but also realized that I think overall it's drifted quite a bit from reality. Let me know if you want to spread the work around, I can update that and push a commit

File: README.md [hunk 5]
```diff
@@ -22,8 +22,6 @@ If you haven't written a post before, please add yourself to `_data/authors.yml`
 
 That's it, we'll take care of the rest. If you wish, you can also submit just a plain Markdown file and we'll be happy to integrate it.
 
-You can also use `tut` in posts. See `posts/2016-09-30-subtype-typeclasses.md` for an example.
```
Comment (samspills) [inline README.md:25]: `tut` is archived, and we don't have any mdoc examples. We can investigate that as a follow up though :)

File: README.md (hunk 3 above)
Comment (samspills) [reply README.md:135]: I updated the readme in [54fcf5c](https://github.com/typelevel/typelevel.github.com/pull/422/commits/54fcf5cd2f2bf92ed89663652221ca9507a6f78c)

File: README.md [hunk 6]
```diff
@@ -125,42 +121,27 @@ Images for styling purposes are located inside `img/`, photos inside `img/media/
...
 
 ```yml
-layout: post
-title: "Cats"
-category: "Functional Programming"
-description: "An experimental library intended to provide abstractions for functional programming in Scala, leveraging its unique features. Design goals are approachability, modularity, documentation and efficiency."
-permalink: "https://non.github.io/cats/"
-github: "https://github.com/non/cats"
+- title: "Cats"
+  description: "An experimental library intended to provide abstractions for functional programming in Scala, leveraging its unique features. Design goals are approachability, modularity, documentation and efficiency."
```
Comment (JasnaMRB) [inline README.md:130]:
```suggestion
  description: "An experimental library that provides abstractions for functional programming in Scala, leveraging its unique features. Design goals are approachability, modularity, and efficiency."
```
optional... I would think "documentation" is sort of a minimum implicit requirement for all projects here and sounds odd to call it a "design goal", though  I acknowledge Cats' docs are among the best--and that this diff in your PR is sorta incidental.

File: README.md [hunk 7]
```diff
@@ -125,42 +121,27 @@ Images for styling purposes are located inside `img/`, photos inside `img/media/
...
-Incubator projects and macros are created a little differently. They are located in `_data/incubator.yml` and `_data/macros.yml` respectively, and look like this:
-
-```yml
-- title: "imp"
-  description: "Summoning implicit values"
-  github: "https://github.com/non/imp"
-```
+To add an:
+- **affiliate** project, add `affiliate: true` to the project entry
+- **core** project, add `core: true` to the project entry
```
Comment (JasnaMRB) [inline README.md:140]:
```suggestion
To add
- an **affiliate** project, add `affiliate: true` to the project entry
- a **core** project, add `core: true` to the project entry
```

File: README.md [hunk 8]
```diff
@@ -125,42 +121,27 @@ Images for styling purposes are located inside `img/`, photos inside `img/media/
...
 
 ### Adding a page
 
-To add a page, create a HTML or Markdown file in the root of the project. The site navigation is not fully dynamic for simplification purposes. It can be changed in the default layout (`_layouts/default.html`).
+To add a page, create a directory in the root of the project, with at least an `index.html` file in that directory. The site navigation is not fully dynamic for simplification purposes. The new page can be added to the nav by updating `_data/nav.yml`.
```
Comment (JasnaMRB) [inline README.md:144]:
```suggestion
To add a page, 

1. Create a directory in the root of the project, with at least an `index.html` file in that directory. 2. Update  `_data/nav.yml` to add it to the navigation. (The site navigation is not fully dynamic for simplification.)
```

File: platforms/index.html [hunk 9]
```diff
@@ -0,0 +1,13 @@
+---
+layout: page
+title: Platforms
+---
+<div id="section-page">
+  <div class="container">
+    <div class="masthead-page">
+      <h1><span>{{ page.title }}</span></h1>
+      <p>Checkout the Typelevel ecosystem by platform</p>
```
Comment (JasnaMRB) [inline platforms/index.html:9]:
```suggestion
      <p>Check out the Typelevel ecosystem by platform</p>
```

**Review by JasnaMRB — APPROVED** [review]
Small non-blocking comments. looks great!

**Review by jducoeur — APPROVED** [review]
I don't think it's an either/or, at least in theory -- personally, I would leave `category` in place, but we should rethink what the breakdown is.

That said, I agree that platform is higher-priority.

File: README.md (hunk 6 above)
Comment (valencik) [reply README.md]: This feedback is good, however, this isn't actually the place to change the description of `cats` that would be in the `_data/projects.yml` file, and I'd like to avoid yak shaving that here as well.

So I've just updated this README snippet to match the current content in the data file.

File: README.md [hunk 10]
```diff
@@ -125,42 +121,29 @@ Images for styling purposes are located inside `img/`, photos inside `img/media/
...
+To add
+- an **affiliate** project, add `affiliate: true` to the project entry
+- a **core** project, add `core: true` to the project entry
 
 ### Adding a page
 
-To add a page, create a HTML or Markdown file in the root of the project. The site navigation is not fully dynamic for simplification purposes. It can be changed in the default layout (`_layouts/default.html`).
+To add a page, 
+
+1. Create a directory in the root of the project, with at least an `index.html` file in that directory. 2. Update  `_data/nav.yml` to add it to the navigation. (The site navigation is not fully dynamic for simplification.)
```
Comment (samspills) [inline README.md:146]:
```suggestion
1. Create a directory in the root of the project, with at least an `index.html` file in that directory. 
2. Update  `_data/nav.yml` to add it to the navigation. (The site navigation is not fully dynamic for simplification.)
```

**Comment by valencik** [comment]
> I don't think it's an either/or, at least in theory -- personally, I would leave `category` in place, but we should rethink what the breakdown is.
> 
> That said, I agree that platform is higher-priority.

I think the category tag could certainly come back in the future if someone gives it some love.
But as is I think it's incomplete, and perhaps noisy on the project cards.
